spring-ai-core/src/main/java/org/springframework/ai/rag/generation/augmentation/ContextualQueryAugmenter.java

```diff
@@ -74,15 +74,15 @@ public final class ContextualQueryAugmentor implements QueryAugmentor {
 			Politely inform the user that you can't answer it.
 			""");
 
-	private static final boolean DEFAULT_ALLOW_EMPTY_CONTEXT = true;
+	private static final boolean DEFAULT_ALLOW_EMPTY_CONTEXT = false;
```

Comment:
This one is very debatable so I'm eager the get some feedback.

When using explicit RAG, I probably want to reduce hallucinations as much as possible. If no similar documents are retrieved from a vector store, I probably want to short-circuit the generation and have the model answering that it doesn't know rather than making stuff up.

I would therefore recommend disallowing having empty context by default. Then the user can decide if they want the model to make stuff up. This should help a lot with troubleshooting and in spotting retrieval errors earlier.

The case where I want to fallback to "standard" generation without context is something that should be handled at the beginning of the RAG flow. There's typically an agentic component that starts a flow by determining whether to send a request directly to the model or go through the retrieval process. It's something I have in my backlog and that will be introduced as a "Scheduling" sub-module.

What do you think about this choice of the default value?

Reply:
Since the main point of advanced rag techniques is better accuracy, i think explicitly dealing with the empty context makes sense, as otherwise it will make up stuff, going against the 'pedal tone' of intent for these classes.